### PR TITLE
Hulya, Isa- Join Existing  List

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person.
Tokens can be shared with other users to allow them to co-manage a given list.

Joining an existing list consists of the following:

* User enters an existing token
* Save the token to `localStorage`
* Show the user the existing list

AC:

* When the user does not already have a token in `localStorage`, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
* Entering the token and hitting submit saves the token to `localStorage`, effectively giving them joint control of the list
* On submit, show an error if the token does not exist
* If they get an error message, allow them to try again or create a new list

Notes:

* As of this story, the welcome/home screen will be feature complete, as reflected in this wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
* For accessibility reasons, use a `<label>` tag for the share token field.
* You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
* If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue 5).

### PR DESCRIPTION
**Pull request by hulyak**
## Description

- If there is no token currently saved in local storage, the user is presented with the form where they can get a new token or they can submit an existing token.
- Users can type an existing token to join an existing list and this token is added to the local storage.
- After the user clicks on Join Existing List, if the token exists, they are presented with the list.
- The items on the list page are the items in the collection that are associated with the token. 
- If the token doesn't exist, the error is shown to the user and they can enter another token using the same form.

## Related Issue

#5 

## Acceptance Criteria

- When the user does not already have a token in localStorage, on the onboarding/home screen, a simple form is displayed that allows the user to enter a token
- Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of the list
- On submit, show an error if the token does not exist
- If they get an error message, allow them to try again or create a new list

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Screen Shot 2021-07-27 at 4 29 46 PM

[](url)

### After

Screen Shot 2021-07-27 at 4 28 37 PM
Screen Shot 2021-07-27 at 4 29 03 PM

## Testing Steps / QA Criteria

- pull down `hi-joinList` branch in your local repository
- `npm start`
- Create a new list and add some items to the list
- Clear the token from local storage and copy the newly created token into the `Share token` input field 
- Check if you can join the existing list

**NOTE:  We cannot use tokens that were created before yesterday to join existing lists. But every new token created here on out should work.**

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -24,9 +44,15 @@ const Home = () => {
...
       <p>-or-</p>
       <p>Join an existing shopping list by entering a three word token.</p>
-      <form>
+      <form onSubmit={verifyToken}>
         <label htmlFor="token">Share token</label>
-        <input type="text" name="token" />
+        <input
+          type="text"
+          name="token"
+          value={sharedToken}
```
Comment (sandaiiyahh) [inline src/components/Home.js:53]: Ahh, Megan told me this previously, but for the htmlFor, it's associated with an "id" property. So how about we have also have an "id" property on our `input` tag set to "token" to connect the `label` with the `input` ?

**Review by sandaiiyahh — COMMENTED** [review]
Hey Hulya and Isa! I just pulled down the branch, cleared my previous token, created and populated a new list, and was able to rejoin it again! Awesome job! I love the use of the state variable with the ternary to show the error message. 

Kelsey and I also used the firestore `.where()` method in our `ItemsList` component, which is cool, so we would probably run into merge conflicts on the `ItemsList` component but nothing big! 🤗

I just left a comment on a minor change you can do regarding the input tag, but other than that, awesome job :))

File: src/components/Home.js (hunk 1 above)
Comment (isaabutaa) [reply src/components/Home.js:53]: That's a great point, Sandy! Totally forgot to add the `id` attribute to the input field. Will do that now. Thanks for suggesting it! 🙏

**Review by sandaiiyahh — APPROVED** [review]

**Comment by ksiman14** [comment]
Hulya and Isa - All works as expected, great job!! I don't have any additional comments at this point.

**Review by ksiman14 — APPROVED** [review]

**Pull request merged**